docker.go

```diff
@@ -164,6 +167,18 @@ func getRoleArnFromEnv(env []string) (role roleArn, policy string, err error) {
 			}
 		} else if v[0] == "IAM_POLICY" && len(v) > 1 {
 			policy = strings.TrimSpace(v[1])
+		} else if v[0] == "IAM_ROLE_NAME" && len(v) > 1 {
```

Comment:
So the valid values for IAM_ROLE should be:
1. `^arn:aws:iam::\d+:role/.+$`   <--- full arn
2. `^(\d+:).+$`   <-- role name with optional account id

If the account id is not provided in the second form, it can be discovered via the metadata API. You would only want to read the account id once, maybe at startup.